src/components/Field/Field.tsx

```diff
@@ -29,6 +29,11 @@ import {CustomizationContext} from '../../components/View/View'
 import {InteractiveInput} from '../../components/ui/InteractiveInput/InteractiveInput'
 import HistoryField from '../../components/ui/HistoryField/HistoryField'
 
+export interface FieldCustomProps {
+    hierarchyDepth?: number,
```

Comment:
This way `Field` still knows about Hierarchy. Custom properties should be of any type and the actual typing for the specific props should be declared in the component that utilizes those props

Reply:
Fixed.